atlasdb-config/src/main/java/com/palantir/atlasdb/http/AtlasDbHttpClients.java

```diff
@@ -43,17 +43,11 @@ private AtlasDbHttpClients() {
     }
 
     /**
-     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, and the
-     * default Feign HTTP client.
+     * Constructs a dynamic proxy for the specified type. Uses a Feign HTTP client with no retrying.
+     *
+     * @param trustContext if present, the proxy will use the supplied SSL factory
+     * @param limitPayloadSize if true, request payload size will be limited
      */
-    public static <T> T createProxy(
-            MetricRegistry metricRegistry,
-            Optional<TrustContext> trustContext,
-            String uri,
-            Class<T> type) {
-        return createProxy(metricRegistry, trustContext, uri, type, UserAgents.DEFAULT_USER_AGENT, false);
-    }
-
     public static <T> T createProxy(
```

Comment:
propagate the name change here, I believe `AtlasDbFeignTargetFactory` isn't used directly in most places, and it comes through here.

Reply:
Seems fair. I'd normally attach the usual warnings of an API break, but _this_ method also seems hardly used outside of Atlas

Reply:
I think we should start employing `@RestrictedApi` so we can at least keep track of things that are atlas only.

Reply:
Updated the method name. As for annotations, shouldn't we just use `@Beta` then?

Reply:
Nah, `@Beta` implies you can use it and it's subject to change, `@RestrictedApi` can fail error prone checks if it's used outside of where we expect it to be used.